lib/mongo/crypt/encryption_io.rb

```diff
@@ -86,6 +86,40 @@ def mark_command(cmd)
 
         return response.first
       end
+
+      # Get information about the AWS encryption key and feed it to the the
+      # KmsContext object
+      #
+      # @param [ Mongo::Crypt::KmsContext ] kms_context A KmsContext object
+      #   corresponding to one AWS KMS data key. Contains information about
+      #   the endpoint at which to establish a TLS connection and the message
+      #   to send on that connection.
+      def feed_kms(kms_context)
```

Comment:
This is my interpretation of the [integration guide](https://github.com/mongodb/libmongocrypt/blob/master/integrating.md#state-mongocrypt_ctx_need_kms). Keep in mind that I know very little about the `Socket::SSL` module in the driver, so feedback would be appreciated 🙏